x/valset-pref/msg_server_test.go

```diff
@@ -292,3 +292,123 @@ func (suite *KeeperTestSuite) TestUnDelegateFromValidatorSet() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestRedelegateValidatorSet() {
+	suite.SetupTest()
+
+	// setup 3 validators
+	valAddrs := suite.SetupMultipleValidators(9)
+
+	tests := []struct {
+		name           string
+		delegator      sdk.AccAddress
+		newPreferences []types.ValidatorPreference
+		coinToStake    sdk.Coin
+		expectPass     bool
+	}{
+		{
+			name:      "update existing Validator set",
+			delegator: sdk.AccAddress([]byte("addr1---------------")),
+			newPreferences: []types.ValidatorPreference{
+				{
+					ValOperAddress: valAddrs[0],
+					Weight:         sdk.NewDecWithPrec(2, 1),
+				},
+				{
+					ValOperAddress: valAddrs[1],
+					Weight:         sdk.NewDecWithPrec(2, 1),
+				},
+				{
+					ValOperAddress: valAddrs[2],
+					Weight:         sdk.NewDecWithPrec(6, 1),
+				},
+			},
+			coinToStake: sdk.NewCoin("stake", sdk.NewInt(20_000_000)),
+			expectPass:  true,
+		},
+		{
+			name:      "update the same validator set as above",
+			delegator: sdk.AccAddress([]byte("addr1---------------")),
+			newPreferences: []types.ValidatorPreference{
+				{
+					ValOperAddress: valAddrs[0],
+					Weight:         sdk.NewDecWithPrec(5, 1),
+				},
+				{
+					ValOperAddress: valAddrs[1],
+					Weight:         sdk.NewDecWithPrec(3, 1),
+				},
+				{
+					ValOperAddress: valAddrs[2],
+					Weight:         sdk.NewDecWithPrec(2, 1),
+				},
+			},
+			coinToStake: sdk.NewCoin("stake", sdk.NewInt(20_000_000)),
+			expectPass:  true,
+		},
+		{
+			name:      "update existing Validator with new validators",
+			delegator: sdk.AccAddress([]byte("addr2---------------")),
+			newPreferences: []types.ValidatorPreference{
+				{
+					ValOperAddress: valAddrs[3],
+					Weight:         sdk.NewDecWithPrec(2, 1),
+				},
+				{
+					ValOperAddress: valAddrs[4],
+					Weight:         sdk.NewDecWithPrec(2, 1),
+				},
+				{
+					ValOperAddress: valAddrs[5],
+					Weight:         sdk.NewDecWithPrec(2, 1),
+				},
+				{
+					ValOperAddress: valAddrs[6],
+					Weight:         sdk.NewDecWithPrec(2, 1),
+				},
+				{
+					ValOperAddress: valAddrs[7],
+					Weight:         sdk.NewDecWithPrec(1, 1),
+				},
+				{
+					ValOperAddress: valAddrs[8],
+					Weight:         sdk.NewDecWithPrec(1, 1),
+				},
+			},
+			coinToStake: sdk.NewCoin("stake", sdk.NewInt(50_000_000)),
+			expectPass:  true,
+		},
```

Comment:
We should also try to  redelegate to a new set, but one of the validators in the old set should be a part of the new set.